PaloAltoCortexXDR/cortex_module/cortex_edr_connector.py

```diff
@@ -68,7 +68,17 @@ def timestamp_cursor(self, time: int) -> None:
 
     @cached_property
     def alert_url(self) -> str:
-        return f"https://api-{self.module.configuration.fqdn}/public_api/v1/alerts/get_alerts_multi_events"
+        base_fqdn = self.module.configuration.fqdn.rstrip("/")
+
+        if base_fqdn.startswith("https://"):
+            if base_fqdn.endswith("/public_api/v1/alerts/get_alerts_multi_events"):
+                return base_fqdn
+            return f"{base_fqdn}/public_api/v1/alerts/get_alerts_multi_events"
+
+        if base_fqdn.startswith("api-"):
+            return f"https://{base_fqdn}/public_api/v1/alerts/get_alerts_multi_events"
+
+        return f"https://api-{base_fqdn}/public_api/v1/alerts/get_alerts_multi_events"
```

Comment:
Can you add tests to check the behavior of this code?
I suggest outsourcing the computation of the alert URL in a function (located in a new file `helpers.py`), to test this function and call it here.

Reply:
Yes i can of course. Thank you.